SamsungPay/src/main/java/com/braintreepayments/api/SamsungPayClient.java

```diff
@@ -0,0 +1,235 @@
+package com.braintreepayments.api;
+
+import androidx.annotation.Nullable;
+import androidx.annotation.VisibleForTesting;
+
+import com.samsung.android.sdk.samsungpay.v2.SpaySdk;
+import com.samsung.android.sdk.samsungpay.v2.payment.CardInfo;
+import com.samsung.android.sdk.samsungpay.v2.payment.CustomSheetPaymentInfo;
+import com.samsung.android.sdk.samsungpay.v2.payment.sheet.CustomSheet;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+
+import static com.braintreepayments.api.SamsungPayMapAcceptedCardBrands.mapToSamsungPayCardBrands;
+import static com.samsung.android.sdk.samsungpay.v2.SpaySdk.SPAY_NOT_READY;
+import static com.samsung.android.sdk.samsungpay.v2.SpaySdk.SPAY_NOT_SUPPORTED;
+import static com.samsung.android.sdk.samsungpay.v2.SpaySdk.SPAY_READY;
+
+public class SamsungPayClient {
+
+    private final BraintreeClient braintreeClient;
+
+    @VisibleForTesting
+    SamsungPayInternalClient internalClient;
+
+    public SamsungPayClient(BraintreeClient braintreeClient) {
+        this.braintreeClient = braintreeClient;
+    }
+
+    /**
+     * Forwards the user to the Samsung Pay update page.
+     * This should be invoked when Samsung Pay returns a {@link SamsungPayError#SAMSUNG_PAY_APP_NEEDS_UPDATE}
+     * error code from {@link SamsungPayClient#isReadyToPay(SamsungPayIsReadyToPayCallback)}
+     */
+    public void goToUpdatePage() {
+        getInternalClient(new GetSamsungPayInternalClientCallback() {
+            @Override
+            public void onResult(@Nullable SamsungPayInternalClient internalClient, @Nullable Exception error) {
+                if (internalClient != null) {
+                    internalClient.goToSamsungPayUpdatePage();
+                    braintreeClient.sendAnalyticsEvent("samsung-pay.goto-update-page");
+                } else {
+                    // TODO: determine if we should notify an error here
```

Comment:
I'd say we callback the error here since it would likely be a configuration error from the internal client right? That would probably be helpful for the merchant

Reply:
Yeah it's just a bit wack since the method is asynchronous for the Braintree SDK, but for SamsungPay itself it's a synchronous operation. I'm down to add a callback.

Reply:
Added callbacks!